src/libraries/System.Net.Quic/tests/FunctionalTests/QuicStreamTests.cs

```diff
@@ -16,204 +17,149 @@ public abstract class QuicStreamTests<T> : QuicTestBase<T>
-                    await connection.ConnectAsync();
                     await using QuicStream stream = connection.OpenBidirectionalStream();
 
                     await stream.WriteAsync(s_data, endStream: true);
+                    await stream.ShutdownWriteCompleted();
 
-                    byte[] memory = new byte[12];
-                    int bytesRead = await stream.ReadAsync(memory);
+                    byte[] buffer = new byte[s_data.Length];
+                    int bytesRead = await ReadAll(stream, buffer);
 
                     Assert.Equal(s_data.Length, bytesRead);
-                    // TODO this failed once...
-                    Assert.True(s_data.Span.SequenceEqual(memory));
-                    await stream.ShutdownWriteCompleted();
-
-                    await connection.CloseAsync(errorCode: 0);
-                });
+                    Assert.True(s_data.Span.SequenceEqual(buffer));
 
-                await (new[] { listenTask, clientTask }).WhenAllOrAnyFailed(millisecondsTimeout: 10000);
-            }
+                    await stream.ShutdownCompleted();
+                }
+            );
         }
 
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/49157")]
         [Fact]
         public async Task MultipleReadsAndWrites()
         {
-            using QuicListener listener = CreateQuicListener();
-
-            for (int j = 0; j < 100; j++)
+            const int sendCount = 5;
+            int expectedBytesCount = s_data.Length * sendCount;
+            byte[] expected = new byte[expectedBytesCount];
+            Memory<byte> m = expected;
+            for (int i = 0; i < sendCount; i++)
             {
-                Task listenTask = Task.Run(async () =>
+                s_data.CopyTo(m);
+                m = m[s_data.Length..];
+            }
+
+            await RunClientServer(
+                iterations: 100,
+                serverFunction: async connection =>
                 {
-                    // Connection isn't being accepted, interesting.
-                    using QuicConnection connection = await listener.AcceptConnectionAsync();
                     await using QuicStream stream = await connection.AcceptStreamAsync();
-                    byte[] buffer = new byte[s_data.Length];
 
-                    while (true)
-                    {
-                        int bytesRead = await stream.ReadAsync(buffer);
-                        if (bytesRead == 0)
-                        {
-                            break;
-                        }
-                        Assert.Equal(s_data.Length, bytesRead);
-                        Assert.True(s_data.Span.SequenceEqual(buffer));
-                    }
+                    byte[] buffer = new byte[expectedBytesCount];
+                    int bytesRead = await ReadAll(stream, buffer);
+                    Assert.Equal(expectedBytesCount, bytesRead);
+                    Assert.True(expected.SequenceEqual(buffer));
```

Comment:
```suggestion
                    Assert.Equal(expected, buffer);
```
I won't repeat it anymore. `Assert` can handle comparison of arrays/enumerables etc. And the message in case of non-equality is slightly better than what `Assert.True` will give you.

Reply:
`Equal` doesn't seem to work on `ReadOnlyMemory<byte>` though... will change them to `byte[]` I guess

Reply:
Interesting, it overrides [`IEquatable<T>`](https://docs.microsoft.com/en-us/dotnet/api/system.readonlymemory-1?view=net-5.0) which should work nicely with xUnit asserts.

Reply:
🤷‍♀️ For this
```c#
byte[] arr = new byte[] { 1, 2, 3 };
ReadOnlyMemory<byte> mem = new byte[] { 1, 2, 3 };

Assert.Equal(arr, mem);
```
the result will be
```
        Assert.Equal() Failure
        Expected: System.ReadOnlyMemory<Byte>[3]
        Actual:   System.ReadOnlyMemory<Byte>[3]
```

Reply:
Baaah, it compares addresses :facepalm:. It totally confused me, I'd expect it to compare the content. Never mind then, do as you consider the best.